app/schema_validation/__init__.py

```diff
@@ -65,39 +65,60 @@ def validate(
     json_to_validate,
     schema,
 ):
+    """Validate a JSON object against a schema.  If the validation fails, log the JSON object with redacted
+    personalisation and ICN information, and raise a ValidationError.
+
+    Args:
+        json_to_validate (dict): The JSON object to validate.
+        schema (dict): The JSON schema to validate against.
+
+    Raises:
+        ValidationError: If the JSON object fails validation.
+
+    Returns:
+        dict: The JSON object with redacted personalisation and ICN information
+    """
+    # Ensure that json_to_validate is a dictionary
+    if not isinstance(json_to_validate, dict):
+        current_app.logger.info('Validation failed for: %s', json_to_validate)
+        errors = [{'error': 'ValidationError', 'message': 'Payload is not json.'}]
```

Comment:
I think you need a more specific error message.  A list, for example, a valid json, but it is not a dictionary.